docs/user_guide/expand_path.rst

```diff
@@ -71,6 +71,221 @@ If your session start time is present in your data path, you can indicate this f
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The IBL's Brain Wide Map features data from several labs of mice performing a visual decision-making task. Some
+experimental sessions, such as those from the Steinmetz Lab, include video recordings of the experiments from three
+cameras, stored in the following directory structure.
+
+.. code-block:: bash
+
+    steinmetzlab/
+    ¦   Subjects/
+    ¦   +-- NR_0017/
+    ¦   ¦   +-- 2022-03-22/
+    ¦   ¦   ¦   +-- 001/
+    ¦   ¦   ¦   ¦   +-- raw_video_data/
+    ¦   ¦   ¦   ¦   ¦   +-- _iblrig_leftCamera.raw.6252a2f0-c10f-4e49-b085-75749ba29c35.mp4
+    ¦   ¦   ¦   ¦   ¦   +-- ...
+    ¦   ¦   ¦   ¦   +-- ...
+    ¦   +-- NR_0019/
+    ¦   ¦   +-- 2022-04-29/
+    ¦   ¦   ¦   +-- 001/
+    ¦   ¦   ¦   ¦   +-- raw_video_data/
+    ¦   ¦   ¦   ¦   ¦   +-- _iblrig_leftCamera.raw.9041b63e-02e2-480e-aaa7-4f6b776a647f.mp4
+    ¦   ¦   ¦   ¦   ¦   +-- ...
+    ¦   ¦   ¦   ¦   +-- ...
+    ¦   ...
+
+We can use :py:class:`~neuroconv.tools.path_expansion.LocalPathExpander` to find these left camera video files and
+extract the subject ID, the session start time (formatted as ``yyyy-mm-dd``), and a hash that we'll treat as
+the session ID (though in actuality it is an object-specific identifier).
+
+.. code-block:: python
+
+    source_data_spec = {
+        "IBL_video": {
+            "base_directory": "/steinmetzlab/",
+            "file_path": "Subjects/{subject_id}/{session_start_time:%Y-%m-%d}/001/raw_video_data/_iblrig_leftCamera.raw.{session_id}.mp4"
+        }
+    }
+    path_expander = LocalPathExpander()
+    metadata_list = path_expander.expand_paths(source_data_spec)
+
+The ``metadata_list`` now contains the information extracted for each matching file found by
+:py:class:`~neuroconv.tools.path_expansion.LocalPathExpander`. The information for the first file is shown below.
+
+.. code-block:: python
+
+    {
+        "source_data": {
+            "IBL_video": {
+                "file_path": "/steinmetzlab/Subjects/NR_0017/2022-03-22/001/raw_video_data/_iblrig_leftCamera.raw.6252a2f0-c10f-4e49-b085-75749ba29c35.mp4"
+            }
+        },
+        "metadata": {
+            "NWBFile": {
+                "session_id": "6252a2f0-c10f-4e49-b085-75749ba29c35",
+                "session_start_time": datetime.datetime(2022, 3, 22, 0, 0)
+            },
+            "Subject": {
+                "subject_id": "NR_0017"
+            }
+        }
+    }
+
+If you would like to experiment locally with :py:class:`~neuroconv.tools.path_expansion.LocalPathExpander`, we
+provide a helper method in ``neuroconv.tools.path_expansion`` that partially replicates the directory structure of
+the IBL data with dummy files on your machine.
+
+.. code-block:: python
+
+    from neuroconv.tools.path_expansion import generate_path_expander_demo_ibl
+
+    generate_path_expander_demo_ibl(folder_path="path/to/generate/dummy/files")
```

Comment:
This will have to be updated

Reply:
ah good catch, completely forgot about that